libs/native-federation-runtime/src/lib/init-federation.ts

```diff
@@ -92,18 +90,43 @@ function processRemoteImports(
 ): Scopes {
   const scopes: Scopes = {};
   const scopedImports: Imports = {};
+  const currentImportMap = importShim.getImportMap();
 
   for (const shared of remoteInfo.shared) {
+    const defaultExternalUrl = getDefaultExternalUrl(currentImportMap, shared);
```

Comment:
#### Do not process host during `initFederation`
Host externals are directly read from `es-module-shims` when processing the remotes